### PR TITLE
Add a recipe to count occurrences of a given string within a message

### PR DESCRIPTION
**Pull request by Data-Meister**
As discussed here: https://coffeescript-cookbook.github.io/chapters/strings/finding-substrings

### CODE REVIEW

**Comment by sukima** [comment]
These three commits can be squashed into one. No need to save the intermediate work in progress with the final merge.

```console
git reset master
git add .
git commit -C 83f146e
```

Or you can do an interactive rebase `git rebase -i`.

**Comment by Data-Meister** [comment]
I edited with the github online editor. Know how to do this with github?

**Comment by lolmaus** [comment]
You can't squash with GitHub web interface. You need to clone your fork repo, checkout your branch and use `git rebase -i HEAD~3` or something like that. Google up squashing.

PS I personally don't mind merging with multiple commits, but apparently this repo has standards.

**Comment by sukima** [comment]
I avoid the GitHub interface except for very small changes. Since the whole concept is to use Git the idea of not using Git is strange and weird to me.

As maintainer of my own projects I usually will volunteer to do the squash myself during the merge. So **I volunteer to do that here as well** But that would mean closing this PR and opening another one. If the rebase were to happen on `Data-Meister:master` branch instead then this PR would autoupdate.

**Comment by lolmaus** [comment]
@sukima If @Data-Meister gives you edit access to his fork repo, you'll be able to force-push while staying in this PR.

**Comment by sukima** [comment]
* Original Commiter rebasing: http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit
* Repo Owner merge squash: http://stackoverflow.com/a/5309051/227176

**Comment by lolmaus** [comment]
> Repo Collaborator merge squash

  
Whoa...

**Comment by sukima** [comment]
@lolmaus you ok with a *Repo Collaborator merge squash*? Means this PR will marked as **closed** not **merged**.

Also,
* https://twitter.com/jamesfublo/status/402407321265274881/photo/1
* http://jamescooke.info/git-to-squash-or-not-to-squash.html

**Comment by lolmaus** [comment]
@sukima Yes, if @Data-Meister retains his credit for the commit.

**Comment by sukima** [comment]
@lolmaus yes the author does not change only the committer.

File: chapters/strings/finding-substrings.md [hunk 1]
```diff
@@ -24,8 +24,14 @@ message.indexOf "This", 5
 message.lastIndexOf "This"
 # => 49
 
-{% endhighlight %}
 
-## Discussion
+# Count occurrences of a given string
+
+    substrCount = (str, subStr) ->
+      re = new RegExp( subStr ,"g" )
```
Comment (lolmaus) [inline chapters/strings/finding-substrings.md]: Would prefer this without brackets:

    re = new RegExp subStr ,"g"

**Comment by sukima** [comment]
Closed by commit 9d67a3999476c412898481273c44ee64a41dcb6c

**Comment by sukima** [comment]
@lolmaus Darn, missed that comment. I don't want to force push to master because I don't know who has pulled since the merge (2 minutes ago). Perhaps another PR?

**Comment by lolmaus** [comment]
I think it's too minor to bother.

@Data-Meister and @sukima, thank you for contributing!

**Comment by Data-Meister** [comment]
You're most welcome @lolmaus and thank you @sukima, I'm new to GitHub and will bear this best-practice in mind for the future.

**Pull request closed without merging**